packages/contracts/tasks/deploy.ts

```diff
@@ -1,60 +1,92 @@
+import { task } from 'hardhat/config';
+import * as fs from 'fs';
 
-type Contracts = "QuadraticFundingCurator" | "QuadraticVoteCollectModule";
+type Contracts = 'QuadraticFundingCurator' | 'QuadraticVoteCollectModule';
 
-task("deploy", "Deploy contracts and verify")
-  .addOptionalParam("grantsRound", "The address of the active Grants Round")
-  .addOptionalParam("lensHub", "The address of the LensHub")
-  .addOptionalParam("moduleGlobals", "The address of ModuleGlobals")
+task('deploy', 'Deploy contracts and verify')
+  .addOptionalParam('grantsRound', 'The address of the active Grants Round')
+  .addOptionalParam('lensHub', 'The address of the LensHub')
+  .addOptionalParam('moduleGlobals', 'The address of ModuleGlobals')
   .setAction(async ({ grantsRound, lensHub, moduleGlobals }, { ethers }) => {
     const [admin] = await ethers.getSigners();
 
     const contracts: Record<Contracts, ContractFactory> = {
-      QuadraticFundingCurator: await ethers.getContractFactory("QuadraticFundingCurator"),
-      QuadraticVoteCollectModule: await ethers.getContractFactory("QuadraticVoteCollectModule"),
+      QuadraticFundingCurator: await ethers.getContractFactory('QuadraticFundingCurator'),
+      QuadraticVoteCollectModule: await ethers.getContractFactory('QuadraticVoteCollectModule'),
     };
 
     const deployments: Record<Contracts, string> = {
-      QuadraticFundingCurator: "",
-      QuadraticVoteCollectModule: "",
+      QuadraticFundingCurator: '',
+      QuadraticVoteCollectModule: '',
     };
 
     //TODO Correct arguments for Mumbai/Polygon
     const constructorArguments: Record<Contracts, string[]> = {
-      QuadraticFundingCurator: ["0x45cf9Ba12b43F6c8B7148E06A6f84c5B9ad3Dd44", admin.address],
+      QuadraticFundingCurator: ['0x45cf9Ba12b43F6c8B7148E06A6f84c5B9ad3Dd44', admin.address],
       QuadraticVoteCollectModule: [
-        "0x60Ae865ee4C725cd04353b5AAb364553f56ceF82",
-        "0x1353aAdfE5FeD85382826757A95DE908bd21C4f9",
+        lensPolygonAddresses.lensHubImplementation,
+        lensPolygonAddresses.moduleGlobals,
       ],
     };
 
+    const constructorArgumentsMumbai: Record<Contracts, string[]> = {
+      QuadraticFundingCurator: ['0x45cf9Ba12b43F6c8B7148E06A6f84c5B9ad3Dd44', admin.address],
+      QuadraticVoteCollectModule: [
+        lensMumbaiAddresses.lensHubImplementation,
+        lensMumbaiAddresses.moduleGlobals,
+      ],
+    };
+
+   const toFile = (path: string, deployment: Record<Contracts, string>) => {
+      fs.writeFileSync(path, JSON.stringify(deployment), { encoding: 'utf-8' });
+    }
+
     for (const [name, contract] of Object.entries(contracts)) {
       console.log(`Starting deployment of ${name}`);
       const factory = contract;
-      const constructorArgs = Object.entries(constructorArguments).find((entry) => entry[0] === name)?.[1];
-      console.log(`Constructor arguments: ${constructorArgs}`);
+      let constructorArgs;
+
+      if (hre.network.name == 'polygon-mainnet') {
```

Comment:
This is kinda iffy. You're repeating the same logic, so it seems like the constructorArguments should be able to handle all networks insteads of creating separate steps

Reply:
Deleted!